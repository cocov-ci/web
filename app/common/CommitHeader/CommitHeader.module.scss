@import "styles/variables";

.base {
  display: flex;
  flex-direction: row;

  a:hover {
    text-decoration: underline;
  }

  .avatarWrapper {
    padding-right: 4px;

    .skeleton > span > span {
      border-radius: 50%;
    }
  }

  .infoWrapper {
    display: flex;
    flex-direction: column;

    .commitMeta {
      > * {
        margin-right: 4px;
      }

      .username {
        color: $colorBlack;
        letter-spacing: 0;
        font-variation-settings: "wght" 500;
      }

      .headLink {
        letter-spacing: 0;
      }
    }

    .commitTitle {
      font-size: 17px;
      letter-spacing: 0;
      font-variation-settings: "wght" 600;
    }
  }
}
