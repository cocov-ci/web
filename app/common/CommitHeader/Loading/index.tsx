import classNames from 'classnames'
import React from 'react'

import Loading from 'app/common/Loading'
import randomBetween from 'utils/randomBetween'

import styles from '../CommitHeader.module.scss'

export type CommitHeaderProps = {
  className?: string
}

const LoadingComponent = ({ className }: CommitHeaderProps) => {
  return (
    <div className={classNames(styles.base, className)}>
      <div className={styles.avatarWrapper}>
        <Loading
          alignment="left"
          className={styles.skeleton}
          height="25px"
          type="skeleton"
          width="25px"
        />
      </div>

      <div className={styles.infoWrapper}>
        <div className={styles.commitMeta}>
          <Loading
            alignment="left"
            height="18px"
            type="skeleton"
            width={`${randomBetween(206, 400)}px`}
          />
        </div>
        <div className={styles.commitTitle}>
          <Loading
            alignment="left"
            height="18px"
            type="skeleton"
            width={`${randomBetween(32, 400)}px`}
          />
        </div>
      </div>
    </div>
  )
}

export default LoadingComponent
