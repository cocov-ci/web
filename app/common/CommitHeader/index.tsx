import classNames from 'classnames'
import React from 'react'

import Avatar from 'app/common/Avatar'
import RelativeTime from 'app/common/RelativeTime'
import { inconsolata } from 'utils/fonts'

import styles from './CommitHeader.module.scss'

type CommitHeaderProps = {
  className?: string
  username: string
  avatarURL: string
  headSHA: string
  headURL: string
  timestamp: Date
  commitMessage: string
}

const CommitHeader = ({
  className,
  username,
  avatarURL,
  headSHA,
  headURL,
  timestamp,
  commitMessage,
}: CommitHeaderProps) => {
  return (
    <div className={classNames(styles.base, className)}>
      <div className={styles.avatarWrapper}>
        <Avatar avatarURL={avatarURL} size="25px" />
      </div>
      <div className={styles.infoWrapper}>
        <div className={styles.commitMeta}>
          <a
            className={styles.username}
            href={`https://github.com/${username}`}
            rel="noreferrer"
            target="_blank"
          >
            {username}
          </a>
          <span>committed</span>
          <a
            className={classNames(styles.headLink, inconsolata.className)}
            href={headURL}
            rel="noreferrer"
            target="_blank"
          >
            {headSHA.substring(0, 6)}
          </a>
          <RelativeTime timestamp={timestamp} />
        </div>
        <div className={styles.commitTitle}>
          <a href={headURL} rel="noreferrer" target="_blank">
            {commitMessage}
          </a>
        </div>
      </div>
    </div>
  )
}

export default CommitHeader
