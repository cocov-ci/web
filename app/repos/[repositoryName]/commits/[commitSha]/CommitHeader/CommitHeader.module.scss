.commitHeader {
  margin: 22px 22px 50px;
}
