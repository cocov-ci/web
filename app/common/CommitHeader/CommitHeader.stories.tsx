import type { Meta, StoryObj } from '@storybook/react'

import CommitHeader from './index'

const meta: Meta<typeof CommitHeader> = {
  title: 'Common/CommitHeader',
  component: CommitHeader,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof CommitHeader>

export const Default: Story = {
  args: {
    headURL: 'https://github.com/cocov-ci/web',
    avatarURL: 'https://avatars.githubusercontent.com/u/118852412?s=200&v=4',
    headSHA: '62bad0cbadbeefc0c0f',
    commitMessage:
      'fix(rollout-restart): Ensure to keep watching rollout status',
    username: 'cocov-ci',
    timestamp: new Date(Date.parse('2022-11-22T21:25:31Z')),
  },
}

export const Readonly: Story = {
  args: {
    headURL: 'https://github.com/cocov-ci/web',
    avatarURL: 'https://avatars.githubusercontent.com/u/118852412?s=200&v=4',
    headSHA: '62bad0cbadbeefc0c0f',
    commitMessage:
      'fix(rollout-restart): Ensure to keep watching rollout status',
    username: 'cocov-ci',
    timestamp: new Date(Date.parse('2022-11-22T21:25:31Z')),
  },
}
