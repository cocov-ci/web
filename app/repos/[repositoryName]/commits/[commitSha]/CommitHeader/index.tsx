import classNames from 'classnames'

import CommitHeader from 'app/common/CommitHeader'
import { HeadProps } from 'types/Commits'

import styles from './CommitHeader.module.scss'

const CommitHeaderComponent = ({
  head,
  repositoryName,
  loading,
  className,
}: {
  head?: HeadProps
  repositoryName?: string
  loading?: boolean
  className?: string
}) => {
  const commitHeader = head &&
    repositoryName && {
      headURL: `https://github.com/${head.org_name}/${repositoryName}/commit/${head.sha}`,
      avatarURL: head.user?.avatar,
      headSHA: head.sha,
      commitMessage: head.message,
      username: head.user?.name || head.author_name,
      isRegisteredUser: Boolean(head.user?.name),
      timestamp: new Date(Date.parse(head.created_at)),
    }

  return (
    <CommitHeader
      className={classNames(styles.commitHeader, className)}
      loading={loading}
      {...commitHeader}
    />
  )
}

export default CommitHeaderComponent
