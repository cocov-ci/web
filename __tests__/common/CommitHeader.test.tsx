import { render, screen } from '@testing-library/react'

import CommitHeader from 'app/common/CommitHeader'

import '@testing-library/jest-dom'

const args = {
  headURL: 'https://github.com/cocov-ci/web',
  avatarURL: 'https://avatars.githubusercontent.com/u/118852412?s=200&v=4',
  headSHA: '62bad0cbadbeefc0c0f',
  commitMessage: 'fix(rollout-restart): Ensure to keep watching rollout status',
  username: 'cocov-ci',
  timestamp: new Date(Date.parse('2022-11-22T21:25:31Z')),
}

describe('common/CommitHeader', () => {
  it('renders CommitHeader', () => {
    render(<CommitHeader {...args} />)

    expect(screen.getByText(args.username)).toBeVisible()
    expect(screen.getAllByRole('link')[0]).toHaveAttribute('href', args.headURL)
    expect(screen.getByText(args.headSHA?.substring(0, 6))).toBeVisible()
    expect(screen.getByText(args.commitMessage)).toBeVisible()
  })

  it('renders CommitHeader with "loading" param', () => {
    const { container } = render(<CommitHeader loading={true} />)

    expect(
      container.getElementsByClassName('react-loading-skeleton').length,
    ).toEqual(3)
  })

  it('renders CommitHeader wth "isRegisteredUser" param', () => {
    const { container } = render(
      <CommitHeader {...args} isRegisteredUser={true} />,
    )

    expect(
      container.querySelector(`[href="https://github.com/${args.username}"]`),
    ).toBeVisible()
  })

  it('renders CommitHeader snapshots', () => {
    const { container } = render(<CommitHeader {...args} />)

    expect(container).toMatchSnapshot()
  })
})
