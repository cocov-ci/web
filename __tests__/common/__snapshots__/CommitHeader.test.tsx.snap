// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/CommitHeader renders CommitHeader snapshots 1`] = `
<div>
  <div
    class="base"
  >
    <div
      class="avatarWrapper"
    >
      <img
        alt=""
        aria-hidden="true"
        class="base"
        src="https://avatars.githubusercontent.com/u/118852412?s=200&v=4"
        style="height: 25px; width: 25px;"
      />
    </div>
    <div
      class="infoWrapper"
    >
      <div
        class="commitMeta"
      >
        <span
          class="username"
        >
          cocov-ci
        </span>
        <span>
          committed
        </span>
        <a
          class="headLink className"
          href="https://github.com/cocov-ci/web"
          rel="noreferrer"
          target="_blank"
        >
          62bad0
        </a>
        <span
          class=""
          title="Nov 22, 2022, 9:25 PM GMT"
        >
          4 months ago
        </span>
      </div>
      <div
        class="commitTitle"
      >
        <a
          href="https://github.com/cocov-ci/web"
          rel="noreferrer"
          target="_blank"
        >
          fix(rollout-restart): Ensure to keep watching rollout status
        </a>
      </div>
    </div>
  </div>
</div>
`;
